include/industry_standard/cxl_tsp.h

```diff
@@ -0,0 +1,306 @@
+#define CXL_TSP_OPCODE_GET_TARGET_CONFIGURATION_RSP 0x04
+#define CXL_TSP_OPCODE_GET_TARGET_CONFIGURATION_REPORT_RSP 0x05
+#define CXL_TSP_OPCODE_LOCK_TARGET_CONFIGURATION_RSP 0x06
+#define CXL_TSP_OPCODE_ERROR_RSP 0x7F
+
+/* Get Target TSP Version */
+
+typedef struct {
+    cxl_tsp_header_t header;
+    uint16_t reserved;
+} cxl_tsp_get_target_tsp_version_req_t;
+
+/* Get Target TSP Version Response */
+
+typedef uint8_t cxl_tsp_version_number_t;
+
+typedef struct {
+    cxl_tsp_header_t header;
+    uint16_t reserved;
+    uint8_t version_number_entry_count;
+    /*cxl_tsp_version_number_t version_number_entry[version_number_entry_count];*/
+} cxl_tsp_get_target_tsp_version_rsp_t;
+
+/* Get Target Capabilities */
+
+typedef struct {
+    cxl_tsp_header_t header;
+    uint16_t reserved;
+} cxl_tsp_get_target_capabilities_req_t;
+
+/* Get Target Capabilities Response */
+
+typedef struct {
+    cxl_tsp_header_t header;
+    uint16_t memory_encryption_features_supported;
+    uint32_t memory_encryption_algorithms_supported;
+    uint16_t memory_encryption_number_of_range_based_keys;
+    uint16_t reserved;
+    uint16_t te_state_change_and_access_control_features_supported;
+    uint16_t reserved2;
+    uint32_t supported_explicit_out_of_band_te_state_granularity;
+    uint32_t supported_explicit_in_band_te_state_granularity;
+    uint16_t configuration_features_supported;
+    uint16_t reserved3;
+    uint32_t number_of_ckids;
+    uint8_t number_of_secondary_sessions;
+    uint8_t reserved4[0x13];
+} cxl_tsp_get_target_capabilities_rsp_t;
+
+#define CXL_TSP_MEMORY_ENCRYPTION_FEATURES_SUPPORT_ENCRYPTION 0x1
+#define CXL_TSP_MEMORY_ENCRYPTION_FEATURES_SUPPORT_CKID_BASED_ENCRYPTION 0x2
+#define CXL_TSP_MEMORY_ENCRYPTION_FEATURES_SUPPORT_RANGE_BASED_ENCRYPTION 0x4
+/* only valid in CAP, not in SET/GET */
+#define CXL_TSP_MEMORY_ENCRYPTION_FEATURES_SUPPORT_INITIATOR_SUPPLIED_ENTROPY 0x8
+/* offset is changed in SET/GET */
+#define CXL_TSP_MEMORY_ENCRYPTION_FEATURES_SUPPORT_CKID_BASED_REQUIRED 0x10
+
+#define CXL_TSP_MEMORY_ENCRYPTION_ALGORITHMS_AES_XTS_128 0x1
+#define CXL_TSP_MEMORY_ENCRYPTION_ALGORITHMS_AES_XTS_256 0x2
+#define CXL_TSP_MEMORY_ENCRYPTION_ALGORITHMS_VENDOR_SPECIFIC 0x80000000
+
+#define CXL_TSP_TE_STATE_CHANGE_AND_ACCESS_CONTROL_FEATURES_SUPPORT_WRITE_ACCESS_CONTROL 0x1
+#define CXL_TSP_TE_STATE_CHANGE_AND_ACCESS_CONTROL_FEATURES_SUPPORT_READ_ACCESS_CONTROL 0x2
+#define CXL_TSP_TE_STATE_CHANGE_AND_ACCESS_CONTROL_FEATURES_SUPPORT_IMPLICIT_TE_STATE_CHANGE 0x4
+#define \
+    CXL_TSP_TE_STATE_CHANGE_AND_ACCESS_CONTROL_FEATURES_SUPPORT_EXPLICIT_OUT_OF_BAND_TE_STATE_CHANGE \
+    0x8
+#define \
+    CXL_TSP_TE_STATE_CHANGE_AND_ACCESS_CONTROL_FEATURES_SUPPORT_EXPLICIT_IN_BAND_TE_STATE_CHANGE \
+    0x10
+#define \
+    CXL_TSP_TE_STATE_CHANGE_AND_ACCESS_CONTROL_FEATURES_SUPPORT_EXPLICIT_TE_STATE_CHANGE_SANITIZE \
+    0x20
+
+#define CXL_TSP_EXPLICIT_OUT_OF_BAND_TE_STATE_CHANGE_GRANULARITY_64B 0x1
```

Comment:
Any single bit is legal. If these are going to be `#define`d it should include all of them.

Reply:
done.